scripts/data_processing/PPI/basemap_mvp.R

```diff
@@ -0,0 +1,300 @@
+#### Base map of population of countries in sjbeckett/subregionalcovid19 at multiple administrative levels ####
+#### Jordan Klein
+
+##### 1) Setup ####
+rm(list = ls())
+
+### Load packages & set authorization for google drive
+library(tidyverse) # data wrangling
+library(tibble) # data wrangling
+library(janitor) # column naming
+library(countrycode) # country c?des
+library(lubridate) # date times
+library(readxl) # excel import
+library(zoo) # calculate rolling averages
+library(R.utils) # R utilities
+library(stringr) # to parse strings in R
+library(dplyr) # data wrangling
+library(readr) # read_csv
+library(stringi)
+library(data.table)
+library(fuzzyjoin)
+library(stringdist)
+library("devtools")
+library("subregionalcovid19")
+googledrive::drive_auth(email = T)
+library(httr)
+library(vietnameseConverter)
+
+#### Read in Pop-Geo file
+Geo.Pop <- st_read("../../../data/raw/GeoPop.shp")
+unique(Geo.Pop$iso3) %>% sort()
+
+#### 2) Get Table of countries' population ####
+##### Geometries
+#### Impot shapefile of country geometries & filter out the 26 used here (-china & taiwan)
+shapefile_raw <- read_delim('https://raw.githubusercontent.com/PandemicPreventionInstitute/NGS-Capacity-map/main/data/Geospatial_Data/geometric_polygons_country.txt', 
+                        delim = "\t") %>% 
+    filter((code %in% unique(Geo.Pop$iso3)) & code != "CHN" & code != "TWN")
+#### Convert into a table of country isocodes & geometries
+shapefile <- select(shapefile_raw, iso3 = code) %>% 
+    st_set_geometry(st_as_sfc(shapefile_raw$geometry, GeoJSON = T))
+
+#### Get geometries for china & taiwan
+shapes_chn_twn <- filter(Geo.Pop, iso3 == "CHN" | iso3 == "TWN") %>%
+    group_by(iso3) %>% 
+    summarise()
+
+### Combine geometries for all 26 countries
+Country_geoms <- bind_rows(shapefile, shapes_chn_twn)
+
+##### Population
+Countries_pop <- st_drop_geometry(Geo.Pop) %>% 
+    group_by(iso3, cntry_n) %>% 
+    summarise(pop = sum(pop, na.rm = T))
+
+##### Combine countries geoms & pop
+Countries <- merge(Countries_pop, Country_geoms, by = "iso3") %>% 
+    st_set_geometry(.$geometry)
+
+### Clear memory
+rm(shapefile, shapefile_raw, shapes_chn_twn)
+gc()
+
+#### 3) Get get level 1 & 2 subnational divisions for each country ####
+##### AUS- only 1st level
+
+##### BEL- have 3rd level, aggregate up to 2nd (treat as 1st)
```

Comment:
Is this what you mean by a canonical admin level 1 -- figuring out for each country what the admin level 1 should be ? Is this standardized somewhere else?

Reply:
These are just notes for what I have in the data.